components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/TopSitesStorage.kt

```diff
@@ -43,15 +43,16 @@ interface TopSitesStorage : Observable<TopSitesStorage.Observer> {
      * If `frecencyConfig` is specified, fill in any missing top sites with frecent top site results.
      *
      * @param totalSites A total number of sites that will be retrieve if possible.
-     * @param fetchProvidedTopSites Whether or not to fetch top sites from the [TopSitesProvider].
      * @param frecencyConfig If [frecencyConfig] is specified, only visited sites with a frecency
      * score above the given threshold will be returned. Otherwise, frecent top site results are
      * not included.
+     * @param providerConfig An instance of [TopSitesProviderConfig] that specifies whether or
+     * not to fetch top sites from the [TopSitesProvider].
      */
     suspend fun getTopSites(
         totalSites: Int,
-        fetchProvidedTopSites: Boolean,
-        frecencyConfig: FrecencyThresholdOption?
+        frecencyConfig: FrecencyThresholdOption? = null,
+        providerConfig: TopSitesProviderConfig? = null
```

Comment:
Should `frecencyConfig` be part of `providerConfig`?

Reply:
Look like`providerConfig` is a generic configuration and `frecencyConfig` one specific type configuration that could be part of the general configuration object.

Reply:
`frecencyConfig` is for auto-generated top sites from the PlacesHistoryStorage and `providerConfig` is for `TopSitesProvider` since both of these storage/provider can be provided separately. I think it makes sense to keep them separate.